File: benchmarks/poly/loopsum/loopsum.go
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	t0 := time.Now()
	acc := 0
	for i := 1; i <= 10000000; i++ {
		acc = (acc + i*i) % 1000000007
	}
	fmt.Println(acc)
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/poly/loopsum/loopsum.py
import sys, time


def main():
    _t0 = time.perf_counter_ns()
    acc = 0
    for i in range(1, 10000001):
        acc = (acc + i * i) % 1000000007
    print(acc)
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()
